x/ccv/parent/client/proposal_handler.go

```diff
@@ -0,0 +1,168 @@
+
+// NewCreateChildChainProposalTxCmd returns a CLI command handler for creating
+// a new consumer chain proposal governance transaction.
+func NewCreateChildChainProposalTxCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "create-child-chain [proposal-file]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Submit a child chain creation proposal",
+		Long: `
+Submit a consumer chain creation proposal along with an initial deposit.
+The proposal details must be supplied via a JSON file.
+
+Example:
+$ %s tx gov submit-proposal create-child-chain <path/to/proposal.json> --from=<key_or_address>
+
+Where proposal.json contains:
+
+{
+    "title": "Create the FooChain",
+    "description": "Gonna be a great chain",
+    "chain_id": "foochain",
+    "initial_height": {
+        "revision_number": 2,
+        "revision_height": 3
+    },
+    "genesis_hash": "Z2VuZXNpcyBoYXNo",
+    "binary_hash": "YmluYXJ5IGhhc2g=",
+    "spawn_time": "2022-01-27T15:59:50.121607-08:00",
+    "deposit": "10000stake"
+}
+		`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			proposal, err := ParseCreateChildChainProposalJSON(args[0])
+			if err != nil {
+				return err
+			}
+
+			content, err := types.NewCreateChildChainProposal(
+				proposal.Title, proposal.Description, proposal.ChainId, proposal.InitialHeight,
+				proposal.GenesisHash, proposal.BinaryHash, proposal.SpawnTime)
+			if err != nil {
+				return err
+			}
+
+			from := clientCtx.GetFromAddress()
+
+			deposit, err := sdk.ParseCoinsNormalized(proposal.Deposit)
+			if err != nil {
+				return err
+			}
+
+			msg, err := govtypes.NewMsgSubmitProposal(content, deposit, from)
+			if err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+}
+
+type CreateChildChainProposalJSON struct {
+	Title         string             `json:"title"`
+	Description   string             `json:"description"`
+	ChainId       string             `json:"chain_id"`
+	InitialHeight clienttypes.Height `json:"initial_height"`
+	GenesisHash   []byte             `json:"genesis_hash"`
+	BinaryHash    []byte             `json:"binary_hash"`
+	SpawnTime     time.Time          `json:"spawn_time"`
+	Deposit       string             `json:"deposit"`
+}
+
+type CreateChildChainProposalReq struct {
+	BaseReq  rest.BaseReq   `json:"base_req"`
+	Proposer sdk.AccAddress `json:"proposer"`
+
+	Title         string             `json:"title"`
+	Description   string             `json:"description"`
+	ChainId       string             `json:"chainId"`
+	InitialHeight clienttypes.Height `json:"initialHeight"`
+	GenesisHash   []byte             `json:"genesisHash"`
+	BinaryHash    []byte             `json:"binaryHash"`
+	SpawnTime     time.Time          `json:"spawnTime"`
+	Deposit       sdk.Coins          `json:"deposit"`
```

Comment:
Use `CreateChildChainProposalJSON` instead.

Reply:
You mean, just embed it?

Reply:
Yeah, to not duplicate code. But it's minor.

Reply:
They are actually different